oss_src/unity/lib/gl_sarray.hpp

```diff
@@ -1721,6 +1721,53 @@ class gl_sarray {
                      flexible_type step = FLEX_UNDEFINED);
 
   /**
+   * Apply an aggregate function over a moving window.
+   * 
+   * \param input The input SArray (expects to be materialized)
+   * \param fn_name string representation of the aggregation function to use.
+   * The mapping is the same string mapping used by the groupby aggregate
+   * function.
+   * \param window_start The start of the moving window relative to the current
+   * value being calculated, inclusive. For example, 2 values behind the current
+   * would be -2, and 0 indicates that the start of the window is the current
+   * value.
+   * \param window_end The end of the moving window relative to the current value
+   * being calculated, inclusive. Must be greater than `window_start`. For
+   * example, 0 would indicate that the current value is the end of the window,
+   * and 2 would indicate that the window ends at 2 data values after the
+   * current.
+   * \param min_observations The minimum allowed number of non-NULL values in the
+   * moving window for the emitted value to be non-NULL. size_t(-1) indicates
+   * that all values must be non-NULL.
+   *
+   * Returns an SArray of the same length as the input, with a type that matches
+   * the type output by the aggregation function.
+   * 
+   * Throws an exception if:
+   *  - window_end < window_start
+   *  - The window size is excessively large (currently hardcoded to UINT_MAX).
+   *  - The given function name corresponds to a function that will not operate
+   *  on the data type of the input SArray.
+   *  - The aggregation function returns more than one non-NULL types.
+   *  
+   *  Example:
+   *  \code
+   *  gl_sarray a{0,1,2,3,4,5,6,7,8,9};
+   *  // Moving window encompasses 3 values behind current and current value.
+   *  auto result = a.rolling_apply(std::string("__builtin__avg__"), -3, 0);
+   *  \endcode
+   *
+   *  Produces an SArray with these values: 
+   *  \code
+   *  {NULL,NULL,NULL,1.5,2.5,3.5,4.5,5.5,6.5,7.5}
+   *  \endcode
+   */
+  gl_sarray rolling_apply(const std::string &fn_name,
```

Comment:
Technically, the gl_sarray is a public interface for the SDK, so I would prefer calling this builtin_rolling_apply (just to future proof our rolling_apply in the future which might accept functions.